src/EnumDefinition.php

```diff
@@ -2,7 +2,8 @@
 
 namespace Spatie\Enum;
 
-class EnumDefinition
+/**  @internal */
+final class EnumDefinition
```

Comment:
I know that @freekmurze doesn't like this. But I've still tried it. 😂
For me there's absolutely no reason to keep this value object open for extension and the `@internal` plus `final` should make this clear.

Reply:
Marking it as `@internal` should be sufficient, as it means users that _do_ want to _juggle with sharp knives_ can expect it to break at any moment, and agree that they take that risk. `final` takes this freedom away.

Reply:
Because of the fact what this class does I absolutely don't see any reason to tinker with this class at all. But am open to remove the `final`.
Do you have any reason in mind to tinker with this class?

Reply:
https://github.com/spatie/enum/pull/64/commits/8607075f3c9c399f5067dbe57c51e0f2fcdd10c4